lib/remap-lang/src/value.rs

```diff
@@ -315,14 +318,26 @@ impl Value {
         let err = || Error::Div(self.kind(), rhs.kind());
 
         let value = match self {
-            Value::Integer(lhv) => (lhv / i64::try_from(&rhs).map_err(|_| err())?).into(),
+            Value::Integer(lhv) => (lhv as f64 / f64::try_from(&rhs).map_err(|_| err())?).into(),
```

Comment:
Is there such a `try_from` function between float and int? I can't find any.

The `f64::try_from(Value` that we are using here is just doing under the hood:

```rust
        match value {
            Value::Integer(v) => Ok(*v as f64),
            Value::Float(v) => Ok(*v),
            _ => Err(Error::Coerce(value.kind(), Kind::Float)),
        }
```

Reply:
I thought there was, but I can't find it now so I must be mistaken. It would have to be emulated:
```rust
    let result = value as f64;
    match result as i64 {
        value => Ok(result),
        _ => Err(Error::Coerce(…)),
    }
```
Again, it might be out of scope for this particular PR, but something to consider making an issue of.

Reply:
I can't see how that emulation would work. Rust won't let you compare an `i64` with an `f64`, so you would have to cast `result` back to an `f64` - by which point the comparison is pointless.

I can't quite see a good way around this, so I think I will close this and raise an issue.

Reply:
That's exactly what this is doing, by casting `value` to `f64` and then back to `i64`, any conversion failure will cause that value to be different than the original.

I agree, though, that should probably go in a new issue.

Reply:
Oh, yes.. Good point. 

I'm not sure that helps though if the number is a power of 2. Take `36028797018963968 as i64`. Converted to a f64 that is `36028797018963970`. But that number converted back to i64 is back to the original `36028797018963968`.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1b03fa41be91569e49e7446e2b87ecff